src/mlpack/methods/ann/layer/conv_layer.hpp

```diff
@@ -125,8 +131,10 @@ class ConvLayer
         arma::Mat<eT> output;
         BackwardConvolutionRule::Convolution(gy.slice(inMap), rotatedFilter,
             output);
-
-        g.slice(outMap) += output;
+        g.slice(outMap) += output.submat(rotatedFilter.n_rows / 2,
```

Comment:
I didn't put an if there because if we don't put a padding the submat would take the whole matrix. But putting if makes that point explicit. Thanks.